src/ServiceControl/MessageRedirects/Api/MessageRedirectsModule.cs

```diff
@@ -0,0 +1,144 @@
+    using ServiceBus.Management.Infrastructure.Nancy.Modules;
+    using ServiceControl.Infrastructure;
+    using ServiceControl.Infrastructure.Extensions;
+    using ServiceControl.MessageRedirects.InternalMessages;
+
+    public class MessageRedirectsModule : BaseModule
+    {
+        public IBus Bus { get; set; }
+
+        public MessageRedirectsModule()
+        {
+
+            Post["/redirects"] = parameters =>
+            {
+                var message = new CreateMessageRedirect
+                {
+                    FromPhysicalAddress = parameters.fromphysicaladdress,
+                    ToPhysicalAddress = parameters.tophysicaladdress
+                };
+
+                if (string.IsNullOrWhiteSpace(message.FromPhysicalAddress) || string.IsNullOrWhiteSpace(message.ToPhysicalAddress))
+                {
+                    return HttpStatusCode.BadRequest;
+                }
+
+                message.MessageRedirectId = DeterministicGuid.MakeId(message.FromPhysicalAddress, message.ToPhysicalAddress);
+
+                using (var session = Store.OpenSession())
+                {
+                    var redirect = session.Load<MessageRedirect>(MessageRedirect.GetDocumentIdFromMessageRedirectId(message.MessageRedirectId));
+
+                    if (redirect != null)
+                    {
+                        return Negotiate.WithReasonPhrase("Duplicate").WithModel(redirect).WithStatusCode(HttpStatusCode.Conflict);
+                    }
+
+                    var dependents = session.Query<MessageRedirect>().Where(r => r.ToPhysicalAddress == message.FromPhysicalAddress).ToList();
+
+                    if (dependents.Any())
+                    {
+                        return Negotiate.WithReasonPhrase("Dependents").WithModel(dependents).WithStatusCode(HttpStatusCode.Conflict);
+                    }
+                }
+
+                Bus.SendLocal(message);
+
+                return HttpStatusCode.Created;
+            };
+
+            Put["/redirects/{messageredirectid}"] = parameters =>
+            {
+                Guid oldMessageRedirectId = parameters.messageredirectid;
+
+                var message = new CreateMessageRedirect
+                {
+                    FromPhysicalAddress = parameters.fromphysicaladdress,
+                    ToPhysicalAddress = parameters.tophysicaladdress
+                };
+
+                if (string.IsNullOrWhiteSpace(message.FromPhysicalAddress) || string.IsNullOrWhiteSpace(message.ToPhysicalAddress))
+                {
+                    return HttpStatusCode.BadRequest;
+                }
+
+                message.MessageRedirectId = DeterministicGuid.MakeId(message.FromPhysicalAddress, message.ToPhysicalAddress);
+
+                Bus.SendLocal(new RemoveMessageRedirect
+                {
+                    MessageRedirectId = oldMessageRedirectId
+                });
+
+                Bus.SendLocal(message);
```

Comment:
The original plan was not to support Update, I think this wasn't wise. We need there to be 1 state for redirects, not 2 different ones. So we will support updates to prevent 2 redirects from the same from queue being available at the same time.

I pulled out the validation routine so it can be used for the update as well.